doc/packages/build-support.md

```diff
@@ -0,0 +1,97 @@
+# Build Support {#sec-build-support}
+
+## `pkgs.substitute` {#pkgs-substitute}
+
+`pkgs.substitute` is a wrapper around [the `substitute` function](#fun-substitute) provided by the
+stdenv. It replaces strings in `src` as specified by the `substitutions` argument.
```

Comment:
"The stdenv" is not a meaningful expression. Either say "the standard environment" if you mean the set of environment variables and Bash functions provided to derivations, or say `stdenv` if your mean that code object. In this case it we should make clear it's about the first. I actually misinterpreted it on the first pass!

```suggestion
`pkgs.substitute` is a wrapper around [the `substitute` Bash function](#fun-substitute) in the standard environment.
It replaces strings in `src` as specified by the `substitutions` argument.
```

Please wrap lines at each sentence, this makes suggestions easier to make for reviews and easier to read for you. This is a general rule in all Nix-related documentation.